src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java

```diff
@@ -80,16 +93,70 @@ public Tenant save(UUID consortiumId, Tenant tenantDto) {
       centralTenantId = getCentralTenantId();
     }
 
-    checkTenantNotExistsAndConsortiumExistsOrThrow(consortiumId, tenantDto.getId());
-    Tenant savedTenant = saveTenant(consortiumId, tenantDto);
+    TenantEntity tenantEntity;
+    if (forceCreatePrimaryAff) {
+      tenantEntity = getOrCreateTenantEntity(consortiumId, tenantDto);
+    }
+    else {
+      checkTenantNotExistsOrThrow(tenantDto.getId());
+      tenantEntity = saveTenantEntity(consortiumId, tenantDto);
+    }
+    var savedTenant = converter.convert(tenantEntity, Tenant.class);
 
     runInFolioContext(createFolioExecutionContextForTenant(tenantDto.getId(), currentTenantContext, folioMetadata),
-      () -> configurationClient.saveConfiguration(createConsortiaConfigurationBody(centralTenantId)));
+        () -> {
+          configurationClient.saveConfiguration(createConsortiaConfigurationBody(centralTenantId));
```

Comment:
with a flag for POST you need to check each operation, so moving to PUT will simplify logic